### DISCUSSION TITLE
How can i write multithread apps with awc

### DISCUSSION THREAD

**Discussion by Necoo33** [actix/actix-web#3266] [Q&A] [answered]
Hello , i'm trying to write some multithread applications which basically will make http requests in multiple threads. I'm tried to use std::thread and awc packages for that but i couldn't make it. Can you write a minimalist example which you do synchronous http requests on multiple threads?

**Comment by robjtede** [comment]
Since `awc::Client` isn't thread-safe, you'll need to create one client per thread.

```rust
std::thread::spawn(|| {
  let client = awc::Client::new();
});
```

**Reply by Necoo33** [reply]
Can i share data between threads with Arc type? If i can't, how can i achieve that kind of behavior?

**Reply by robjtede** [reply, marked as answer]
Depends what data. You can't share the client itself with an Arc, nor Request/Response types. but a response **body** would be Send-able.

If you want complete share-ability then I'd point you toward a thread-safe client like `reqwest`.

**Reply by Necoo33** [reply]
Thanks. It'd be very good if you implement this in future releases.

**Reply by robjtede** [reply]
Being thread-local is part of the feature set of Actix Web generally thought I agree it makes less sense for the client part and it will certainly be considered for future releases. Appreciate your input 👍🏻
